File: backend/src/entities/Task.ts
import { Entity, PrimaryGeneratedColumn, Column, CreateDateColumn, UpdateDateColumn, DeleteDateColumn, ManyToOne } from "typeorm";
import { Company } from "./Company";

@Entity()
export class Task {
  @PrimaryGeneratedColumn("uuid")
  id!: string;

  @Column({ nullable: true })
  title?: string;

  @Column({ type: "text", nullable: true })
  body?: string;

  @Column({ default: "TODO" })
  status: string = "TODO";

  @Column({ nullable: true, type: "datetime" })
  dueAt?: Date;

  @CreateDateColumn()
  createdAt!: Date;

  @UpdateDateColumn()
  updatedAt!: Date;

  @DeleteDateColumn()
  deletedAt?: Date;

  @ManyToOne(() => Company, (company) => company.tasks)
  company?: Company;
}

File: server/app/models/task.py
from typing import Optional
from beanie import Document, Indexed, Link
from datetime import datetime
from .company import Company
from .person import Person

class Task(Document):
    title: Indexed(str)
    description: Optional[str] = None
    due_date: Optional[datetime] = None
    priority: str = "Medium" # Low, Medium, High, Urgent
    status: str = "Todo" # Todo, In Progress, Completed, Archived
    related_to_type: Optional[str] = None # company, deal, lead, person
    related_to_id: Optional[str] = None
    owner_id: Optional[str] = None # User UUID
    created_at: datetime = datetime.utcnow()
    updated_at: datetime = datetime.utcnow()

    class Settings:
        name = "tasks"
